pkg/server/util/regentryutil/fetch.go

```diff
@@ -12,7 +12,9 @@ import (
 
 func FetchRegistrationEntries(ctx context.Context, dataStore datastore.DataStore, spiffeID string) ([]*common.RegistrationEntry, error) {
 	fetcher := newRegistrationEntryFetcher(dataStore)
-	return fetcher.Fetch(ctx, spiffeID)
+	// Enable ListRegistrationEntries and GetNodeSelectors operations to use readOnly DB connections by enabling tolerateStale.
+	// If the server is not configured with readOnly DB connections, the reads are defaulted to primary readWrite DB instances.
+	return fetcher.Fetch(ctx, spiffeID, true)
```

Comment:
Instead of passing true here and threading `tolerateStale` through each method, I think it's probably simpler to just set `TolerateStale: true` on the ListRegistrationEntries and GetNodeSelectors calls.

I don't see us making it configurable anytime soon and this code is harder to audit.

Reply:
Sure, addressed in the latest diff!